src/appleseed/renderer/kernel/shading/shadingpoint.cpp

```diff
@@ -33,6 +33,7 @@
 // appleseed.renderer headers.
 #include "renderer/kernel/intersection/intersector.h"
 #include "renderer/modeling/input/source.h"
+#include "renderer/modeling/input/sourceshadingpoint.h"
```

Comment:
Same question in all the .cpp files where this is the only change.

Reply:
An implicit conversion to shadingpoint pod class happens in all those files, in evaluate() calls. Without that headers conversion and compilation fails. Is implicit conversion good or bad? I'm not sure.